src/radical/pilot/task_description.py

```diff
@@ -426,13 +425,17 @@ class TaskDescription(ru.TypedDict):
             during service startup.
 
         timeout (float, optional): Any timeout larger than 0 will result in
-            the task process to be killed after the specified amount of seconds.
-            The task will then end up in `CANCELED` state.
-
-        startup_timeout (float, optional): This setting is specific for service
-            tasks: any value larger than 0 will abort the service if after that
-            time no service information has been obtaines, i.e., if the service
-            startup takes too long.  The service will end up in `FAILED` state.
+            the task process to be killed after the specified amount of seconds,
+            starting from the launch time, if startup_timeout is not set, or
+            starting from execution start time, if startup_timeout is set and
+            was tracked. The task will then end up in `CANCELED` state.
+
+        startup_timeout (float, optional): Any value larger than 0 will abort
+            the task process if it hasn't started the actual execution process,
+            e.g., batch system didn't provide allocated resources or the service
+            startup takes too long. This option is very important for service
+            tasks. The service task will end up in `FAILED` state, regular task
+            will end up in `CANCELED` state.
```

Comment:
Maybe we should always move to `CANCELED` on timeout for better consistency?

Reply:
with service tasks it's a little different - while task is failed (killing service process), Agent_0 raises `RuntimeError`, thus agent terminates itself. Should I keep description as it is, but adding info about agent?

Reply:
Good point.  I would argue that the agent should not fail on service failures - those are reported to the application after all and can be acted upon.  That would be out of scope for this PR though - should we open an issue for that?